### PR TITLE
Add rename ingest processor

### PR DESCRIPTION
**Pull request by gaobinlong**
### Description
Add documentation for the existing rename ingest processor, this documentation needs to be backported to all versions.

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/6950

### Checklist
- [x] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by vagimeli — APPROVED** [review]
Doc review complete. Minimal edits.

File: _ingest-pipelines/processors/rename.md [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+title: Rename
+parent: Ingest processors
+nav_order: 230
+redirect_from:
+   - /api-reference/ingest-apis/processors/rename/
+---
+
+# Rename processor
+
+The `rename` processor is used to rename an existing field, which can also be used to move a field from one object to another object or the root level.
```
Comment (natebower) [inline _ingest-pipelines/processors/rename.md:12]:
```suggestion
The `rename` processor is used to rename an existing field, which can also be used to move a field from one object to another object or to the root level.
```

File: _ingest-pipelines/processors/rename.md [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+The following table lists the required and optional parameters for the `rename` processor.
+
+| Parameter  | Required/Optional  | Description  |
+|---|---|---|
+`field`  | Required  | The field name containing the data to be removed. Supports [template snippets]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets). |
+`target_field`  | Required  | The new name of the field. Supports [template snippets]({{site.url}}{{site.baseurl}}/ingest-pipelines/create-ingest/#template-snippets). |
+`ignore_missing`  | Optional  | Specifies whether the processor should ignore documents that do not contain the specified `field`. If set to `true`, the processor does not modify the document if the `field` does not exist. Default is `false`. |
+`description`  | Optional  | A brief description of the processor.  |
+`if` | Optional | A condition for running the processor. |
+`ignore_failure` | Optional | Specifies whether the processor continues execution even if it encounters errors. If set to `true`, failures are ignored. Default is `false`. |
```
Comment (natebower) [inline _ingest-pipelines/processors/rename.md:39]:
```suggestion
`ignore_failure` | Optional | Specifies whether the processor continues execution even if it encounters an error. If set to `true`, failures are ignored. Default is `false`. |
```

File: _ingest-pipelines/processors/rename.md [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+`on_failure` | Optional | A list of processors to run if the processor fails. |
+`tag` | Optional | An identifier tag for the processor. Useful for debugging in order to distinguish between processors of the same type. |
+
+## Using the processor
+
+Follow these steps to use the processor in a pipeline.
+
+**Step 1: Create a pipeline** 
+
+The following query creates a pipeline named `rename_field` that moves a field in a object to the root level:
```
Comment (natebower) [inline _ingest-pipelines/processors/rename.md:49]:
```suggestion
The following query creates a pipeline named `rename_field` that moves a field in an object to the root level: 
```

**Review by natebower — COMMENTED** [review]
@gaobinlong @vagimeli Please see my changes and let me know if you have any questions. Thanks!

**Pull request merged**